clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -287,7 +289,37 @@ void executeCommandWithRetry(
         return;
       }
       LOGGER.info("Sending command {} of topic: {}, partition: {} to fork process.", command, topicName, partition);
-      if (remoteCommandSupplier.get()) {
+      try {
+        if (command.equals(START_CONSUMPTION)) {
+          // Start consumption should set up resource ingestion status for tracking purpose.
+          getMainIngestionMonitorService().setVersionPartitionToIsolatedIngestion(topicName, partition);
+        }
+        if (remoteCommandSupplier.get()) {
+          return;
+        }
+      } catch (Exception e) {
+        if (command.equals(START_CONSUMPTION)) {
+          // Failure in start consumption request should reset the resource ingestion status.
+          LOGGER.warn("Clean up ingestion status for topic: {}, partition: {}.", topicName, partition);
+          getMainIngestionMonitorService().cleanupTopicPartitionState(topicName, partition);
+        }
+        throw e;
+      }
+      /**
+       * The idea of this check below is to add resiliency to isolated ingestion metadata management.
+       * Although in most of the case the resource ingestion status managed by main / forked process should be in sync,
+       * but in event of regression or unexpected error metadata could be out of sync.
+       * This extra check covers the case where resource is maintained locally but main process think it is in forked
+       * process and thus keeps failing. If the check indicates that resource is managed locally, it will execute the
```

Comment:
If you are referring to the comment chunk I post between 309-313: It is not happening, it is just extra safeguard I add in case regression leads to metadata in sync.